server/src/controller/products/deleteProduct.ts

```diff
@@ -0,0 +1,20 @@
+import { Request, Response } from 'express';
+import { Product } from '../../database';
+import CustomizeError from '../../error/customizeError';
+import handleKnownExceptions from '../../error/handleKnownError';
+
+const deleteProduct = async (req: Request, res: Response) => {
```

Comment:
here u need to add the IReqUser that I made previously because the express request doesn't include the user added in middleware.